crates/polars-core/src/series/mod.rs

```diff
@@ -304,12 +304,15 @@ impl Series {
         if !dtype.is_known() || (dtype.is_primitive() && dtype == self.dtype()) {
             return Ok(self.clone());
         }
-        let ret = self.0.cast(dtype);
+        #[cfg(feature = "dtype-categorical")]
+        if let DataType::Enum(None, _) = dtype {
+            polars_bail!(ComputeError: "can not cast / initialize Enum without categories present")
+        }
```

Comment:
You're fixing a issue with the Python constructor in the `cast` method on the Rust side. It might work, but I don't think this is the way to go.